### DISCUSSION TITLE
SockJS support or alternative for browser clients with fallback

### DISCUSSION THREAD

**Discussion by mbchangellc** [rabbitmq/rabbitmq-server#11255] [Questions]
Does RabbitMQ support SockJS?  I've seen some docs that say yes but an issue on the SockJS GitHub says no.  Is there an alternative, recommended, mechanism for connecting browser and mobile clients that supports fallback where WebSocket isn't available, i.e. XHR streaming, long polling, etc.  We've looked at socket.io but that seems to require we build a gateway to RabbitMQ ourselves.  Our backend is Java and our frontends are Flutter.  For background, we're building a financial services application that needs to support users on restricted networks which may not allow WebSocket.  We can't just run over port 443 because some of them use proxies that terminate SSL and inspect the traffic, etc.  Thanks for your help.

**Comment by PabloCurty** [comment]
Same here.

**Comment by michaelklishin** [comment]
@mbchangellc it used to but SockJS support was dropped years ago. 

[MQTT-over-WebSockets](https://www.rabbitmq.com/docs/web-mqtt) and [STOMP-over-WebSockets](https://www.rabbitmq.com/docs/web-stomp) are the two primary options for Web-based clients.
